README.md

```diff
@@ -1 +1,182 @@
+
+- **registry**: The on-chain smart contract that powers the Scaffold Stellar registry system, enabling contract verification, naming, and version management.
+
+
+## Features
+
+- **CLI Plugins for Stellar**
+  - `stellar scaffold`: Initialize and manage Scaffold Stellar projects
+    - Creates smart contract projects with best practices
+    - Includes frontend setup using [scaffold-stellar-frontend](https://github.com/AhaLabs/scaffold-stellar-frontend)
+  - `stellar registry`: Publish Wasm binaries and deploy smart contracts
+  - Automated development workflow with hot reloading
+
+  Currently these are available as separate binaries: `stellar-scaffold` and `stellar-registry` respectively.
+
+- **Declarative Environment Management**
+  - Environment-specific builds (development, testing, staging, production)
+  - Seamless integration with both local and deployed contracts
+  - Network configuration via `environments.toml`
+
+- **Coming soon: Smart Contract Registry**
+  - On-chain publishing platform for Wasm binaries
+  - Version management and contract naming
+  - Contract verification and dependency management
+
+- **Coming soon: Deployment Pipeline**
+  - Streamlined deployment process for testnet and mainnet
+  - Contract lifecycle management
+  - Automated environment updates
+
+## Project Structure
+
+- `stellar-scaffold-cli`: Main CLI tool for project scaffolding and development
+- `stellar-registry-cli`: Contract registry and deployment management
+- `stellar-build`: Build utilities for Stellar smart contracts
+- `stellar-scaffold-macro`: Procedural macros for contract development
+
+## Installation
+
+### Development Setup
+```bash
+just setup && just build
+```
+
+### Direct Installation
+To install the executables directly:
+
+```bash
+# Install stellar-scaffold-cli
+cargo install --git https://github.com/ahalabs/scaffold-stellar stellar-scaffold-cli
+
+# Install stellar-registry-cli
+cargo install --git https://github.com/ahalabs/scaffold-stellar stellar-registry-cli
+```
+
+## Quick Start
+
+1. Install the required CLI tools:
+```bash
+# Install scaffold-stellar CLI
+cargo install --git https://github.com/ahalabs/scaffold-stellar scaffold-stellar-cli
+
+# Install registry CLI (needed for deployments)
+cargo install --git https://github.com/ahalabs/scaffold-stellar stellar-registry-cli
+```
+
+2. Initialize a new project:
+```bash
+stellar-scaffold init my-project
+cd my-project
+```
+
+3. Set up your development environment:
+```bash
+# Copy and configure environment variables
+cp .env.example .env
+
+# Install frontend dependencies
+npm install
+```
+
+4. Start development environment:
+```bash
+# Terminal 1: Start the development environment with hot reloading
+# This builds contracts and generates TypeScript clients
+stellar-scaffold dev --build-clients
+
+# Terminal 2: Start the frontend development server
+npm run dev
```

Comment:
- We should make `npm run dev` also run `stellar scaffold dev --build-clients`
- We should make `--build-clients` the default! I don't know why @willemneal keeps fighting me on this 😄